theseus/core/tests/test_objective.py

```diff
@@ -275,10 +303,65 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
         v2_data = torch.ones(batch_size, dof) * f2
         objective.update({"v1": v1_data, "v2": v2_data})
         error = objective.error()
+        _check_error_for_data(v1_data, v2_data, error, "error")
         error_norm_2 = objective.error_squared_norm()
 
         assert error.shape == (batch_size, 2 * dof)
-        _check_error_for_data(v1_data, v2_data, error, error_norm_2)
+        _check_error_for_data(v1_data, v2_data, error_norm_2, "error_norm_2")
+
+        v1_data_new = torch.ones(batch_size, dof) * f1 * 0.1
+        v2_data_new = torch.ones(batch_size, dof) * f2 * 0.1
+
+        input_data = {"v1": v1_data_new, "v2": v2_data_new}
+        error = objective.error(input_data=input_data, also_update=False)
+
+        _check_error(
+            v1_data_new,
+            v2_data_new,
+            error,
+            "error",
+            objective,
+            input_data,
+            also_update=False,
+        )
+
+        error_norm_2 = objective.error_squared_norm(
+            input_data=input_data, also_update=False
```

Comment:
Can we use a different value of `v1_data_new` and `v2_data_new` when passing to `error_squared_norm`, so it's not the same as the one that used for `error()`?